lib/carbon/http.py

```diff
@@ -1,74 +1,21 @@
-from twisted.internet import defer, protocol, reactor
-from twisted.web import client
-from twisted.web.http_headers import Headers
-from twisted.web.iweb import IBodyProducer
-from zope.interface import implements
-
-try:
-  from urllib import urlencode
-except ImportError:
-  from urllib.parse import urlencode
-
-
-class QuietHTTP11ClientFactory(client._HTTP11ClientFactory):
-    noisy = False
-
+import urllib3
 
 # async http client connection pool
-pool = client.HTTPConnectionPool(reactor)
-pool._factory = QuietHTTP11ClientFactory
-
-
-class StringProducer(object):
-  implements(IBodyProducer)
-
-  def __init__(self, body):
-    self.body = body
-    self.length = len(body)
-
-  def startProducing(self, consumer):
-    consumer.write(self.body)
-    return defer.succeed(None)
-
-  def pauseProducing(self):
-    pass
-
-  def stopProducing(self):
-    pass
-
-
-class SimpleReceiver(protocol.Protocol):
-  def __init__(self, response, d):
-    self.response = response
-    self.buf = ''
-    self.d = d
-
-  def dataReceived(self, data):
-    self.buf += data
-
-  def connectionLost(self, reason):
-    # TODO: test if reason is twisted.web.client.ResponseDone, if not, do an errback
-    self.d.callback({
-      'code': self.response.code,
-      'body': self.buf,
-    })
+http = urllib3.PoolManager()
 
 
-def httpRequest(url, values=None, headers=None, method='POST'):
-  fullHeaders = {
-    'Content-Type': ['application/x-www-form-urlencoded']
-  }
-  if headers:
-    fullHeaders.update(headers)
+def httpRequest(url, values=None, headers=None, method='POST', timeout=5):
+  try:
+    result = http.request(
+      method,
+      url,
+      fields=values,
+      headers=headers,
+      timeout=timeout)
+  except BaseException as err:
+    raise Exception("Error requesting %s: %s" % (url, err))
 
-  def handle_response(response):
-    d = defer.Deferred()
-    response.deliverBody(SimpleReceiver(response, d))
-    return d
+  if result.status != 200:
```

Comment:
maybe >= 300 ?

Reply:
Graphite will return 200, why do you think it should be more permissive?